### ISSUE TITLE
auxiliary loss

### ISSUE THREAD

**Issue by lds92** [hszhao/PSPNet#59]
Hello! I am trying to create my own training script for PSPNet, and I do not understand how to apply the auxiliary loss. 
I have to do another pyramid pooling for res4b22, then softmax and sum them????
If someone can help me, I will appreciate it :)

**Comment by krishnavishalv** [comment]
After reading the paper, I found the loss function has two softmax loss functions addedup together with the weight of 0.4.
Take pspnet101_VOC2012_473.prototxt as an example, I guess the loss function should be added to conv6(not conv6_interp ) and conv4_23 * 0.4, L= conv6 + 0.4*conv4_23, then softmax with the label. But the training will be difficult, it will need multi-gpu training because the batch size is big.

**Comment by zhipeng-fan** [comment]
@krishnavishalv @hszhao  What's the label here? it will be the segmentation map or a class in pre-train stage?

**Comment by dheerajpai** [comment]
Can you also explain why is it necessary?

**Comment by littletomatodonkey** [comment]
> @krishnavishalv @hszhao What's the label here? it will be the segmentation map or a class in pre-train stage?

aux layer和最后的output layer的feature size大小都是相同的，因此也是经过softmax之后与最后的label进行比较，具体可以参见gluoncv中PSPNet的实现方法。
